cmd/monaco/integrationtest/account/deploy-download_test.go

```diff
@@ -0,0 +1,129 @@
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/cmd/monaco/runner"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/featureflags"
+	stringutils "github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/strings"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/persistence/account/loader"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"strings"
+	"testing"
+)
+
+func TestE2E(t *testing.T) {
+	t.Setenv(featureflags.AccountManagement().EnvName(), "true")
+	createMZone(t)
+
+	toID := stringutils.Sanitize
+
+	type (
+		given struct {
+			project string
+		}
+
+		test struct {
+			name  string
+			given given
+		}
+	)
+
+	deploy := func(project string, fs afero.Fs) *account.Resources {
+		command := fmt.Sprintf("account deploy --project %s --verbose", project)
+		printCommand(command)
+		cli := runner.BuildCli(fs)
+		cli.SetArgs(strings.Split(command, " "))
+
+		err := cli.Execute()
+		require.NoError(t, err)
+
+		r, err := loader.Load(fs, project)
+		require.NoError(t, err)
+
+		return r
+	}
+	download := func(project string, fs afero.Fs) *account.Resources {
+		command := fmt.Sprintf("account download --project %s --output-folder output --verbose", project)
+		printCommand(command)
+		cli := runner.BuildCli(fs)
+		cli.SetArgs(strings.Split(command, " "))
+
+		err := cli.Execute()
+		require.NoError(t, err)
+
+		r, err := loader.Load(fs, fmt.Sprintf("%s/%s/%s", "output", project, "test-account"))
+		require.NoError(t, err)
+
+		return r
+	}
+
+	tests := []test{
+		{
+			name: "test",
+			given: given{
+				project: "add_user",
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			baseFs := afero.NewCopyOnWriteFs(afero.NewBasePathFs(afero.NewOsFs(), "resources/e2e"), afero.NewMemMapFs())
+			randomizeConfiguration(t, baseFs, tc.given.project)
+			baseFs = afero.NewReadOnlyFs(baseFs)
+
+			deploy1st := deploy(tc.given.project, baseFs)
+			downlo1st := download(tc.given.project, afero.NewCopyOnWriteFs(baseFs, afero.NewMemMapFs()))
+
+			for _, u := range deploy1st.Users {
+				assert.Contains(t, downlo1st.Users, u.Email)
+			}
+			for _, p := range deploy1st.Policies {
+				p.ID = toID(p.Name)
+				assert.Contains(t, downlo1st.Policies, p.ID)
+			}
+			for _, g := range deploy1st.Groups {
+				g.ID = toID(g.Name)
+				assert.Contains(t, downlo1st.Groups, g.ID)
+			}
+
+			deploy2nd := deploy(tc.given.project, baseFs)
+			downlo2nd := download(tc.given.project, afero.NewCopyOnWriteFs(baseFs, afero.NewMemMapFs()))
+			assert.Equal(t, deploy2nd, deploy1st)
+
+			for _, u := range deploy1st.Users {
+				assert.Equal(t, downlo1st.Users[u.Email], downlo2nd.Users[u.Email])
+			}
+			for _, p := range deploy1st.Policies {
+				p.ID = toID(p.Name)
+				assert.Equal(t, downlo1st.Policies[p.ID], downlo2nd.Policies[p.ID])
```

Comment:
nit: isn't `downlo1st.Policies[p.ID]` already available as the loop variable `p`?